File: DSA 4th Year/Gfg POTD/Prajwal/23_.py
def is_vowel(char):
    vowels = 'aeiou'
    return char in vowels

def solve(S):
    count = 0
    n = len(S)

    for i in range(n):
        vowels = set()
        consonants = set()

        for j in range(i, n):
            if is_vowel(S[j]):
                vowels.add(S[j])
            else:
                consonants.add(S[j])

            if len(vowels) == len(consonants):
                count += 1

    return count

# Test the function
if __name__ == "__main__":
    num_cases = int(input())

    # Custom test cases
    test_cases = []
    for _ in range(num_cases):
        test_case = input()
        test_cases.append(test_case)

    # Test the function
    for test_case in test_cases:
        print(solve(test_case))

File: DSA 4th Year/Gfg POTD/Prajwal/23.cpp
#include <iostream>
#include <string>
#include <set>
using namespace std;

bool is_vowel(char c) {
    return (c == 'a' || c == 'e' || c == 'i' || c == 'o' || c == 'u');
}

int solve(string S) {
    int count = 0;
    int n = S.length();
    
    for (int i = 0; i < n; i++) {
        for (int j = i; j < n; j++) {
            set<char> vowels;
            set<char> consonants;
            for (int k = i; k <= j; k++) {
                if (is_vowel(S[k])) {
                    vowels.insert(S[k]);
                } else {
                    consonants.insert(S[k]);
                }
            }
            if (vowels.size() == consonants.size()) {
                count++;
            }
        }
    }
    return count;
}

int main() {
    int T;
    cin >> T;
    cin.ignore(); // ignore the newline character after integer input
    while (T--) {
        string S;
        getline(cin, S);
        cout << solve(S) << endl;
    }
    return 0;
}
